### PR TITLE
[dev-v2.8] Add active remotedialer connections dashboard to rancher-monitoring's "Performance debugging" dashboard

### PR DESCRIPTION
**Pull request by aruiz14**
## Problem

We normally use this graph when troubleshooting Rancher performance problems, and I recurrently find it annoying having to visually compare existing Add/Removed graphs to figure out if the number of active connections is abnormal or not, to the point I normally just go to the Prometheus console and run that query there directly.

## Solution

Add a new panel which directly presents the number of active connections directly.

Screenshot 2024-03-14 at 13 03 04

I structured the commits in this PR by different steps:
  - Updates the dashboard schema by running the Rancher and running the current chart version, then saving the JSON model for this dashboard.
  - Changed RemoteDialer added and removed panels to use a rate, as suggested by the query builder.
  - Added a new panel for Active connections, by showing the difference between added and removed.
  - Finally, run `make` commands to generate all the files.
 
## Testing

Chart is installed normally, Grafana includes a pre-built "Performance debugging" dashboard, and it has the desired panels mentioned above.

### Regressions Considerations

There are 2 dashboards that were changed, and they won't show absolute numbers anymore, but that raw data is still available in the Prometheus query console.

## Backporting considerations

- [dev-v2.9 PR](https://github.com/rancher/charts/pull/3762)

### CODE REVIEW

**Comment by joshmeranda** [comment]
@aruiz14 can you add some more context here? I'm unclear on why these changes are needed.

Are there any issues / jira tickets linked to this?

**Comment by aruiz14** [comment]
> can you add some more context here? I'm unclear on why these changes are needed.

Hi @joshmeranda, we normally use this graph when troubleshooting Rancher performance problems, and I recurrently find it annoying having to visually compare both Add/Removed graphs to figure out if the number of active connections is abnormal or not, to the point I normally just go to the Prometheus console and run that query there directly.

This change is just an attempt to (in my opinion) improve the graph by just adding this new panel. Also, with the presence of this panel, then it's makes more sense to change Add/Remove graphs to use a rate instead of the absolute number (as suggested by the Grafana query builder itself).

The commit refreshing the schema is purely to eliminate noise from the other 2 commits and make them easier to review, since Grafana would include those changes anyway when exporting the dashboard's JSON model.

> Are there any issues / jira tickets linked to this?

I didn't create any, since it comes from my own experience using this dashboard and I believe it's a trivial change. Do I need to create one?

Thanks

**Review by lucasmlp — APPROVED** [review]
LGTM 🚀

**Comment by aruiz14** [comment]
@lucasmlp @joshmeranda it seems GitHub won't allow me to merge this PR. Will you do that?
Thanks!

**Comment by recena** [comment]
@aruiz14 Because the [owner](https://github.com/rancher/charts/blob/dev-v2.8/.github/CODEOWNERS#L62) must review and approve it.

**Comment by aruiz14** [comment]
@joshmeranda since you already participated in this PR and seem to be part of the owners team, could you please take a look? I'll rerun the `make` commands to update it.

**Comment by aruiz14** [comment]
@alexandreLamarre @mallardduck could you please take a look, since you seem to be part of Observation&Backup team and you [recently worked in changes in the same chart](https://github.com/rancher/charts/pull/3656). The CI used to pass but after the last rebase, the `Check Container Images` started failing. Do you know why it happens and how to fix it?
Note that this PR only intents to apply a small modification in one of the predefined Grafana dashboards, nothing else.
Thanks!

**Review by moio — APPROVED** [review]
LGTM

**Comment by moio** [comment]
@prachidamle can we get a review from code owners please?

**Comment by alexandreLamarre** [comment]
@moio @aruiz14 this is going to take a while to review, I can leave some high level feedback now though. We don't intend to get this into the next 2.8 release

**Review by alexandreLamarre — CHANGES_REQUESTED** [review]
First of all this looks like a great improvement, thanks for taking the initiative!

A couple of housekeeping things:
- We'll need an issue in rancher/rancher for this improvement
- A PR should be opened against dev-v2.9 first
- Each section of the PR template should be filled out, where applicable, we can assist there if needed

Next up, something went very wrong with either resolving git conflicts or the patch process in the last commit. It is removing many things that should not be touched. It is also modifying base repositories away from our mirrored repository and air gap setups, among other things.

To fix that, I suggest dropping ALL commits here, and working from the ground up.

To recap the development process, we should:

- run `PACKAGE=rancher-monitoring make prepare`
- make relevant changes for this patch (i.e. grafana dashboards) only to files present in` packages/rancher-monitoring`
- run  `PACKAGE=rancher-monitoring/<subchart> make patch`
- run `PACKAGE=rancher-monitoring make clean`
- commit only the files left over after the patch, with an informative name
- run `PACKAGE=rancher-monitoring make charts`, and commit the files changed with this operation as `make charts`

Once that process is completed, we'll more thoroughly review the changes made to the dashboards. If you have any questions or need clarification at any step, feel free to reach out to us

**Comment by aruiz14** [comment]
Thanks for your comment @alexandreLamarre 

> * We'll need an issue in rancher/rancher for this improvement

✅ Created https://github.com/rancher/rancher/issues/45075

> * A PR should be opened against dev-v2.9 first

✅ Created https://github.com/rancher/charts/pull/3762

> * Each section of the PR template should be filled out, where applicable, we can assist there if needed

✅ Done, used here and in the new `dev-v2.9` PR. Please let me know if there is anything I should change. 

> Next up, something went very wrong with either resolving git conflicts or the patch process in the last commit. It is removing many things that should not be touched. It is also modifying base repositories away from our mirrored repository and air gap setups, among other things.

You are right, something definitely looked wrong, possibly due to using the wrong `PACKAGE` argument when calling `make`. I've followed now your instructions and the result looks much nicer, thanks.
However, since I'm changing only an "overlay file", I noticed that running prepare + _changes_ + patch + clean would only modify one single file that is already in the git working tree, so I took the chance to split that change into multiple commits, for clarity and easing the review.

> @moio @aruiz14 this is going to take a while to review, I can leave some high level feedback now though. We don't intend to get this into the next 2.8 release

I'll respect whatever decision you take here, just please note that the changes are really tiny, but very noisy due to update in the Grafana schema. I recommend to look at each commit separately for easier review:
https://github.com/rancher/charts/pull/3627/commits

Thanks!

**Comment by alexandreLamarre** [comment]
@aruiz14  Thanks, I will take a look at both these PRs soon

**Comment by aruiz14** [comment]
@alexandreLamarre since the feature freeze date was extended a little bit, could we please consider including this change into the next 2.8 release? Thanks!

**Comment by alexandreLamarre** [comment]
@aruiz14 The bottleneck here is QA bandwidth and with the amount of changes going into the next release, I don't think we can fit this in unfortunately

**Comment by moio** [comment]
Honest question: does this need to go through QA at all?

AFAIU the Performance debugging dashboard is by developers, for developers - hidden behind the `CATTLE_PROMETHEUS_METRICS` environment variable which is not documented.

**Comment by aruiz14** [comment]
I made a small improvement based on this comment:
https://github.com/rancher/rancher/issues/45129#issuecomment-2061368666

**Comment by aruiz14** [comment]
Friendly ping @alexandreLamarre.
I've just rebuilt the `make` commands to resolve conflicts created since the last update.

**Comment by aruiz14** [comment]
> I've just rebuilt the `make` commands to resolve conflicts created since the last update.

Once again

**Review by alexandreLamarre — APPROVED** [review]
:shipit:
